helix-rest/src/main/java/org/apache/helix/rest/metadatastore/accessor/ZkRoutingDataWriter.java

```diff
@@ -62,10 +71,12 @@ public ZkRoutingDataWriter(String namespace, String zkAddress) {
     }
 
     // Get the hostname (REST endpoint) from System property
-    // TODO: Fill in when Helix REST implementations are ready
-    ZNRecord myServerInfo = new ZNRecord("dummy hostname");
+    ZNRecord myServerInfo = new ZNRecord(
+        System.getProperty(MetadataStoreRoutingConstants.HOSTNAME_SYSTEM_PROPERTY_KEY));
```

Comment:
What if the property is not set in system?

Reply:
+1. We could probably throw an IllegalStateException here if System.getProperty returns null.

Reply:
Good point. Fixing.